web/src/poll_modal.ts

```diff
@@ -21,33 +21,33 @@ function add_option_row(e) {
     create_option_row($(e.target));
 }
 
-function delete_option_row(e) {
+function delete_option_row(e: {target: JQuery<HTMLInputElement>}): void {
     const $row = $(e.target).closest(".option-row");
     $row.remove();
 }
 
-export function poll_options_setup() {
+export function poll_options_setup(): void {
     const $poll_options_list = $("#add-poll-form .poll-options-list");
     const $submit_button = $("#add-poll-modal .dialog_submit_button");
     const $question_input = $("#add-poll-form #poll-question-input");
 
     // Disable the submit button if the question is empty.
     $submit_button.prop("disabled", true);
     $question_input.on("input", () => {
-        if ($question_input.val().trim() !== "") {
+        const inputValue = $question_input.val();
+        if (inputValue !== null && typeof inputValue === "string" && inputValue.trim() !== "") {
```

Comment:
You shouldn't need to add if clauses for these values not being strings. `$<HTMLInputElement>` is helpful, but also I think `$question_input.val()!` might be more appropriate here; it'll be a bug that should throw an exception if one of these values is even not a string.

Reply:
Hi, I tried the same approach but the issue is. The `$question_input` can be `string | number | string[]` and trim doesnt exist on any other type except `string`.

Reply:
- There was one solution i.e to use `.toString()` but I thought. `$submit_button.prop("disabled", false);` depends on this, therefore its a bad idea to use it.
- another solution could be to use `.text()` or `.html()` at line 32. But I am not sure if thats what the function's need.

So at the end, checking it using the if else is the best I could come up for without breaking the previous logic.
Although, I am open for any suggestions

Reply:
I expect `.text` would be a reasonable solution. @LaPulgaaa may be able to help with some of these more basic questions.

Reply:
`$question_input` is attribute of a `input` element.
```
        <input type="text" id="poll-question-input" class="modal_text_input" placeholder="{{t 'Your question'}}" />
```
`.text()` is used to retrieve or set text content of `HTML` element(like `div`, `span`, `p`) not the value attribute of a `form` element like `input`. 
 `.val()` is specifically used for that purpose. I think using `.toString()` might be a better approach.

Reply:
Hi,
I made the requested changes.
Thanks